agent/main.go

```diff
@@ -90,6 +92,33 @@ func setupflags() {
 	for _, hiddenFlag := range hiddenFlags {
 		_ = pflag.CommandLine.MarkHidden(hiddenFlag)
 	}
+	feature.MutableGates.AddFlag(pflag.CommandLine)
+}
+
+func handleHostRegistration(k8sClient client.Client, hostName string, logger logr.Logger) (err error) {
+	if feature.Gates.Enabled(feature.SecureAccess) {
+		logger.Info("secure access enabled, waiting for host to be registered by ByoAdmission Controller")
+	} else {
+		registration.LocalHostRegistrar = &registration.HostRegistrar{K8sClient: k8sClient}
+		err := registration.LocalHostRegistrar.Register(hostName, namespace, labels)
+		return err
+	}
+	return nil
+}
+
+func setupTemplateParser() *cloudinit.TemplateParser {
+	var templateParser *cloudinit.TemplateParser
+	if feature.Gates.Enabled(feature.SecureAccess) {
```

Comment:
Sure. Curiosity question, what is the advantage of one over the other?

Reply:
Not really. But setting up template parser doesn't care about feature flag. Setting the feature flag causes registrar to be nil and because registrar is nil, you cannot set template parser field.

If suppose you have to add more conditionals around registrar, you will have to repeat that condition here. 
But this is a minor suggestion though. Feel free to disagree :)